openedx/features/discounts/views.py

```diff
@@ -77,12 +81,15 @@ def get(self, request, course_key_string):
         payload = {'discount_applicable': discount_applicable, 'discount_percent': discount_percent}
 
         # Record whether the last basket loaded for this course had a discount
-        ExperimentData.objects.update_or_create(
-            user=request.user,
-            experiment_id=REV1008_EXPERIMENT_ID,
-            key='discount_' + str(course),
-            value=discount_applicable
-        )
+        try:
+            ExperimentData.objects.update_or_create(
+                user=request.user,
+                experiment_id=REV1008_EXPERIMENT_ID,
+                key='discount_' + str(course),
+                value=discount_applicable
+            )
+        except Exception as e:  # pylint: disable=broad-except
+            log.exception(str(e))
```

Comment:
if i'm not mistaken, all of this code was used for the auto applying discount that we are not using, so i think it can be deleted?

Reply:
this is correct. However, the lift to actually delete all of the things that need to be deleted is much more than just patching this. :)

Reply:
to be clear I was only suggesting deleting the ExperimentData object